watchtower-plugin/src/net/http.rs

```diff
@@ -154,14 +155,14 @@ pub async fn send_appointment(
 pub async fn post_request<S: Serialize>(
     endpoint: &str,
     data: S,
-    proxy: Option<String>,
+    proxy: &Option<ProxyInfo>,
 ) -> Result<Response, RequestError> {
     let url = reqwest::Url::parse(endpoint).map_err(|e| {
         RequestError::ConnectionError(format!("Cannot connect to the given URL. {}", e))
     })?;
     let client = if url.host_str().unwrap().ends_with(".onion") {
         if let Some(proxy) = proxy {
-            let proxy = reqwest::Proxy::http(format!("socks5h://{}", proxy))
+            let proxy = reqwest::Proxy::http(proxy.get_socks_addr())
                 .map_err(|e| RequestError::ConnectionError(format!("{}", e)))?;
             reqwest::Client::builder()
                 .proxy(proxy)
```

Comment:
So this checks if the host is a `.onion` url and uses the proxy for it.

But we don't make use of the `always_use_proxy` option flag that CLN provides. We should prioritize that first, i.e.: if `always_use_proxy` flag is set and the url isn't a `.onion` we should **still** use the proxy, as not using it in such a case will expose the node's IP address.

Something along the lines of:
```rust
if let Some(proxy) = proxy {
	if proxy.always_use || url.is_onion() {
		// construct a proxied client.
	} else {
		// construct a non-proxied client.
	}
} else {
	if url.is_onion() {
		return Err("url is onion but no proxy")
	}
	// construct a non-proxid client.
}
```

Reply:
This should be covered now, alongside some minor improvements.

I've added an endpoint param to some of the http functions. A followup would be nice defining the tower endpoints as an enum and replacing all string endpoint with the proper variant (I'm already working on this).